cmd/influx/task.go

```diff
@@ -3,342 +3,458 @@ package main
+		id, err := influxdb.IDFromString(b.org.id)
 		if err != nil {
 			return err
 		}
 		filter.OrganizationID = id
 	}
 
-	if taskFindFlags.limit < 1 || taskFindFlags.limit > influxdb.TaskMaxPageSize {
+	if b.taskFindFlags.limit < 1 || b.taskFindFlags.limit > influxdb.TaskMaxPageSize {
 		return fmt.Errorf("limit must be between 1 and %d", influxdb.TaskMaxPageSize)
 	}
-	filter.Limit = taskFindFlags.limit
+	filter.Limit = b.taskFindFlags.limit
 
 	var tasks []*influxdb.Task
 
-	if taskFindFlags.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.id)
+	if b.taskID != "" {
+		id, err := influxdb.IDFromString(b.taskID)
 		if err != nil {
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return b.printTasks(taskPrintOpts{tasks: tasks})
+}
+
+type taskRerunFailedFlags struct {
+	before      string
+	after       string
+	executeRuns bool
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedCmd() *cobra.Command {
+	cmd := b.opts.newCmd("retry-failed", b.taskRetryFailedF, true)
+	cmd.Short = "Retry failed runs"
+	cmd.Aliases = []string{"rtf"}
+
+	b.org.register(b.opts.viper, cmd, false)
+	b.globalFlags.registerFlags(b.opts.viper, cmd)
+	registerPrintOptions(b.opts.viper, cmd, &b.taskPrintFlags.hideHeaders, &b.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&b.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.before, "before", "", "runs before this time")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.after, "after", "", "runs after this time")
+	cmd.Flags().BoolVar(&b.taskRerunFailedFlags.executeRuns, "run", false,
+		"actually execute failed runs; defaults false")
+
+	return cmd
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedF(*cobra.Command, []string) error {
+	if err := b.org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+	tskSvc, _, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+
+	var failedRuns []*influxdb.Run
+	if b.taskID == "" {
+		failedRuns, err = b.getFailedRunsForOrg()
+	} else {
+		failedRuns, err = b.getFailedRunsForTaskID()
+	}
+	if err != nil {
+		return err
+	}
+
+	for _, run := range failedRuns {
+		if b.taskRerunFailedFlags.executeRuns {
+			newRun, err := tskSvc.RetryRun(context.Background(), run.TaskID, run.ID)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Retry for task %s's run %s queued as run %s.\n", run.TaskID, run.ID, newRun.ID)
+		} else {
+			fmt.Printf("Would retry for %s run for Task %s.\n", run.ID, run.TaskID)
+		}
+
+	}
+	return nil
+
+}
```

Comment:
Awesome, thanks for adding the dry run capability.

I think it would be good to add some additional logging around this. Specifically, if we're doing a dry run, a message informing the user how to do a real run would be helpful. Something like this at the end:

```
Dry run complete. Found %d tasks to retry. Re-run with '--run' to execute.
```

Reply:
Do the structs returned by `getFailedRunsForX()` include the error message of the failure? It could be nice to include in the logs.